src/main/java/seedu/address/model/project/Project.java

```diff
@@ -16,10 +15,10 @@ public class Project {
     private String issue;
 
     /**
-     * Every field must be present and not null.
+     * Name field must be present and not null and other fields may be optional.
      */
     public Project(Name name, Repository repository, Deadline deadline, String client, String issue) {
-        requireAllNonNull(name, repository, deadline, client, issue);
+        requireNonNull(name);
```

Comment:
Is it possible to change back to null checks for all parameters but create singleton default objects for each of the parameters as discussed in the meeting?